pr_agent/algo/utils.py

```diff
@@ -341,3 +343,11 @@
     if user_labels:
         get_logger().info(f"Keeping user labels: {user_labels}")
     return user_labels
+
+def get_max_tokens(model):
+    max_tokens_model = MAX_TOKENS[model]
+    if get_settings().config.max_model_tokens:
+        if max_tokens_model > get_settings().config.max_model_tokens:
+            max_tokens_model = get_settings().config.max_model_tokens
+            # get_logger().debug(f"limiting max tokens to {max_tokens_model}")
+    return max_tokens_model
```

Comment:
**Suggestion:** Add a guard clause to 'get_max_tokens' to handle the case where the model is not found in the MAX_TOKENS dictionary, which would currently raise a KeyError.
```suggestion
def get_max_tokens(model):
    if model not in MAX_TOKENS:
        raise ValueError(f"Model '{model}' not found in MAX_TOKENS dictionary.")
    max_tokens_model = MAX_TOKENS[model]
    settings = get_settings()
    if settings.config.max_model_tokens and max_tokens_model > settings.config.max_model_tokens:
        max_tokens_model = settings.config.max_model_tokens
        # get_logger().debug(f"limiting max tokens to {max_tokens_model}")
    return max_tokens_model
```